mosspy/moss.py

```diff
@@ -69,40 +69,46 @@ def setDirectoryMode(self, mode):
     def setExperimentalServer(self, opt):
         self.options['x'] = opt
 
-    def addBaseFile(self, file):
-        self.basefiles.append(file)
+    def addBaseFile(self, file_path, display_name=None):
+        if os.path.isfile(file_path):
+            self.base_files.append((file_path, display_name))
+        else:
+            raise Exception("addBaseFile({}) => File Not Found".format(file_path))
 
-    def addFile(self, file):
-        if os.path.isfile(file):
-            self.files.append(file)
+    def addFile(self, file_path, display_name=None):
+        if os.path.isfile(file_path):
+            self.files.append((file_path, display_name))
         else:
-            raise Exception("addFile({}) => File Not Found".format(file))
+            raise Exception("addFile({}) => File Not Found".format(file_path))
 
     def addFilesByWildcard(self, wildcard):
         for file in glob.glob(wildcard):
-            self.files.append(file)
+            self.files.append((file, None))
 
     def getLanguages(self):
-        return languages
+        return self.languages
+
+    def uploadFile(self, s, file_path, display_name, file_id):
+        if display_name is None:
+            # If no display name added by user, default to file path
+            display_name = os.path.basename(file_path).replace(" ", "_")
```

Comment:
@nareddyt I tested code against the following directory hierarchy and got into same issue stated in #3.  
```
submission\
submission\p146011\a01.py
submission\p146099\a01.py
```
I think, this `basename` needs to be removed? What do you say...

Reply:
Ahh yes, let me change that